arangod/Transaction/Methods.cpp

```diff
@@ -1901,7 +1901,7 @@ Future<OperationResult> transaction::Methods::modifyLocal(
     Result res = collection->getPhysical()->lookupKeyForUpdate(
         this, key.stringView(), lookupResult);
     if (res.fail()) {
-      if (res.is(TRI_ERROR_ARANGO_CONFLICT)) {
+      if (res.is(TRI_ERROR_ARANGO_CONFLICT) && !isArray) {
```

Comment:
In insertLocal we have a similar check where we also consider the replication type:
https://github.com/arangodb/arangodb/blob/e858626cc7eef181f3366aa11810963754f1c019/arangod/Transaction/Methods.cpp#L1460-L1461
Shouldn't we do the same here? If yes, then this would also apply to the check in line 1947.